core/src/main/java/org/kiwix/kiwixmobile/core/search/DefaultAdapter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.search;
+
+import android.content.Context;
+import android.text.Html;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.TextView;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.recyclerview.widget.RecyclerView;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import org.kiwix.kiwixmobile.core.R;
+
+import static javax.swing.UIManager.getString;
+import static org.kiwix.kiwixmobile.core.utils.StyleUtils.dialogStyle;
+
+public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
+
+  private Context context;
+  private AdapterView.OnItemClickListener onItemClickListener;
+  List<String> searchList;
+  @Inject
```

Comment:
this will not work, have you tested the code?

Reply:
I did not get you. The butterknife?

Reply:
`@Inject` is dagger code, you do not call inject with a component in this class so you cannot use field injection. The android team have finally published a [guide on this topic](https://developer.android.com/training/dependency-injection) I doubt this code would even compile, have you run it on a device? Is your pre-commit hook running? It would have caught this and prevented a commit

Butterknife is annotation based too but uses different ones `@BindView` primarily. Read more [here](https://jakewharton.github.io/butterknife/)

Reply:
Are you talking about this warning? 
```
w: warning: Supported source version 'RELEASE_6' from annotation processor 'org.jetbrains.kotlin.kapt3.base.ProcessorWrapper' less than -source '1.8'
```

I think I am a bad programmer :-(  I didn't need most of these variables.
I could get rid of OnItemClickListener, and even searchPresenter

Reply:
Nobody starts out writing good code, quite the opposite. You just have to keep trying over and over.

Dagger is a huge topic to get into but one of its "benefits" is it has compile time safety. Maybe it won't trigger on this because a binding has been supplied for SearchPresenter but this variable _will_ be null at runtime because just annotating a field `@Inject` does nothing

Reply:
Oh! I got it. I am just getting rid of these variables.